src/Analyzers/CSharp/Analyzers/UsePatternCombinators/CSharpUsePatternCombinatorsDiagnosticAnalyzer.cs

```diff
@@ -38,23 +38,29 @@ protected override void InitializeWorker(AnalysisContext context)
         private void AnalyzeNode(SyntaxNodeAnalysisContext context)
         {
             var expression = (ExpressionSyntax)context.Node;
-
+            if (expression.GetDiagnostics().Any(diagnostic => diagnostic.Severity == DiagnosticSeverity.Error))
```

Comment:
so my test case had an error on `i` (instance field access in instance field initializer), I'm surprised that this didn't catch it..